splice_machine/src/main/java/com/splicemachine/derby/stream/function/AbstractFileFunction.java

```diff
@@ -186,6 +195,14 @@ else if (values.size() < returnRow.nColumns()) {
                         else
                             dvd.setValue(SpliceDateFunctions.TO_TIMESTAMP(value, timestampFormat),calendar);
                         break;
+                    case StoredFormatIds.SQL_CHAR_ID:
+                    case StoredFormatIds.SQL_VARCHAR_ID:
+                        dvd.setValue(value);
+                        //normalize the char type
+                        if(dataTypeDescriptors != null && !dvd.isNull()){
+                            dvd.normalize(dataTypeDescriptors[i-1], dvd);
+                        }
+                        break;
```

Comment:
If we want to catch truncation errors, perhaps other types like SQL_LONGVARCHAR_ID and SQL_CLOB_ID should be handled here as well.

Would it be worthwhile to build a NormalizeResultSetNode into the tree?  It seems like that would automatically handle all necessary normalizations for all data types.  It may add some processing overhead though.

Reply:
I added the check for SQL_CLOB_ID.
After looking at SQLLongvarchar's case, I think we don't need to do normalization here. Different from the generic normalization operation which tries to convert from any type to any type, here we are just trying to convert String to the defined type. For SQLLongvarchar, it does not specify any column length, so putting string into the dvd of SQLLongvarchar type is not an issue.